src/librustc_parse/parser/mod.rs

```diff
@@ -1,16 +1,17 @@
 pub mod attr;
+mod diagnostics;
 mod expr;
+mod generics;
 mod item;
+mod nonterminal;
 mod pat;
 mod path;
-mod ty;
-pub use path::PathStyle;
-mod diagnostics;
-mod generics;
 mod stmt;
-use diagnostics::Error;
+mod ty;
 
 use crate::lexer::UnmatchedBrace;
+use diagnostics::Error;
+pub use path::PathStyle;
```

Comment:
Are there any word of mouth and/or written guidelines about the ordering of mods and uses? I've fixed this negatively, by not mixing uses and modules in random order, but i am wondering what the positive fix would look like. In rust-analyer, we are somewhat picky about this: https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#order-of-imports

Reply:
I don't think there's an official guideline, but the most common order is
```
use crate::things;

use other_crate::things;

use std::things;

mod foo;

/* Other items */
```